--- a/drivers/mfd/mt6397-core.c	2026-09-27 16:55:53.226208479 -0500
+++ b/drivers/mfd/mt6397-core.c	2026-09-27 16:55:32.916126629 -0500
--- a/drivers/mfd/mt6397-core.c
+++ b/drivers/mfd/mt6397-core.c
@@ -123,6 +123,21 @@
 	DEFINE_RES_MEM(MT6323_PWRC_BASE, MT6323_PWRC_SIZE),
 };
 
+static const struct resource mt6359_gauge_resources[] = {
+	DEFINE_RES_IRQ_NAMED(MT6359_IRQ_FG_BAT_H, "COULOMB_H"),
+	DEFINE_RES_IRQ_NAMED(MT6359_IRQ_FG_BAT_L, "COULOMB_L"),
+	DEFINE_RES_IRQ_NAMED(MT6359_IRQ_BAT2_H, "VBAT_H"),
+	DEFINE_RES_IRQ_NAMED(MT6359_IRQ_BAT2_L, "VBAT_L"),
+	DEFINE_RES_IRQ_NAMED(MT6359_IRQ_NAG_C_DLTV, "NAFG"),
+	DEFINE_RES_IRQ_NAMED(MT6359_IRQ_BATON_BAT_OU, "BAT_OUT"),
+	DEFINE_RES_IRQ_NAMED(MT6359_IRQ_FG_ZCV, "ZCV"),
+	DEFINE_RES_IRQ_NAMED(MT6359_IRQ_FG_N_CHARGE_L, "FG_N_CHARGE_L"),
+	DEFINE_RES_IRQ_NAMED(MT6359_IRQ_FG_IAVG_H, "FG_IAVG_H"),
+	DEFINE_RES_IRQ_NAMED(MT6359_IRQ_FG_IAVG_L, "FG_IAVG_L"),
+	DEFINE_RES_IRQ_NAMED(MT6359_IRQ_BAT_TEMP_H, "BAT_TMP_H"),
+	DEFINE_RES_IRQ_NAMED(MT6359_IRQ_BAT_TEMP_L, "BAT_TMP_L"),
+};
+
 static const struct mfd_cell mt6323_devs[] = {
 	{
 		.name = "mt6323-rtc",
@@ -251,6 +266,12 @@
 		.num_resources = ARRAY_SIZE(mt6359_accdet_resources),
 		.resources = mt6359_accdet_resources,
 	},
+	{
+		.name = "mt6359-gauge",
+		.num_resources = ARRAY_SIZE(mt6359_gauge_resources),
+		.resources = mt6359_gauge_resources,
+		.of_compatible = "mediatek,mt6359-gauge",
+	},
 };
 
 static const struct mfd_cell mt6397_devs[] = {
