--- a/include/linux/mfd/mt6397/rtc.h	2026-09-27 16:55:55.441319010 -0500
+++ b/include/linux/mfd/mt6397/rtc.h	2026-09-27 16:55:35.316261047 -0500
@@ -48,6 +48,8 @@
 #define RTC_OFFSET_COUNT       7
 
 #define RTC_AL_SEC             0x0018
+#define RTC_AL_HOU             0x001c
+#define RTC_AL_MTH             0x0022
 
 #define RTC_AL_SEC_MASK        0x003f
 #define RTC_AL_MIN_MASK        0x003f
@@ -60,11 +62,26 @@
 #define RTC_PDN2               0x002e
 #define RTC_PDN2_PWRON_ALARM   BIT(4)
 
+#define RTC_SPAR0              0x0030
+
 #define MTK_RTC_POLL_DELAY_US  10
 #define MTK_RTC_POLL_TIMEOUT   (jiffies_to_usecs(HZ))
 
+#define SPARE_REG_WIDTH        1
+
+enum mtk_rtc_spare_enum {
+	SPARE_AL_HOU,
+	SPARE_AL_MTH,
+	SPARE_SPAR0,
+#ifdef SUPPORT_PWR_OFF_ALARM
+	SPARE_KPOC,
+#endif
+	SPARE_RG_MAX,
+};
+
 struct mtk_rtc_data {
 	u32                     wrtgr;
+	const struct reg_field *spare_reg_fields;
 };
 
 struct mt6397_rtc {
@@ -76,6 +93,7 @@
 	int                     irq;
 	u32                     addr_base;
 	const struct mtk_rtc_data *data;
+	struct regmap_field     *spare[SPARE_RG_MAX];
 };
 
 #endif /* _LINUX_MFD_MT6397_RTC_H_ */
+++ b/drivers/rtc/rtc-mt6397.c	2026-09-27 16:55:31.805063767 -0500
@@ -16,6 +16,15 @@
 #include <linux/mfd/mt6397/rtc.h>
 #include <linux/mod_devicetable.h>
 
+static const struct reg_field mtk_rtc_spare_reg_fields[SPARE_RG_MAX] = {
+	[SPARE_AL_HOU] = REG_FIELD(RTC_AL_HOU, 8, 15),
+	[SPARE_AL_MTH] = REG_FIELD(RTC_AL_MTH, 8, 15),
+	[SPARE_SPAR0] = REG_FIELD(RTC_SPAR0, 0, 7),
+#ifdef SUPPORT_PWR_OFF_ALARM
+	[SPARE_KPOC]   = REG_FIELD(RTC_PDN1, 14, 14),
+#endif
+};
+
 static int mtk_rtc_write_trigger(struct mt6397_rtc *rtc)
 {
 	int ret;
@@ -37,6 +46,49 @@
 	return ret;
 }
 
+static int rtc_nvram_read(void *priv, unsigned int offset, void *val,
+							size_t bytes)
+{
+	struct mt6397_rtc *rtc = dev_get_drvdata(priv);
+	unsigned int ival;
+	int ret;
+	u8 *buf = val;
+
+	mutex_lock(&rtc->lock);
+
+	for (; bytes; bytes--) {
+		ret = regmap_field_read(rtc->spare[offset++], &ival);
+		if (ret)
+			goto out;
+		*buf++ = (u8)ival;
+	}
+out:
+	mutex_unlock(&rtc->lock);
+	return ret;
+}
+
+static int rtc_nvram_write(void *priv, unsigned int offset, void *val,
+							size_t bytes)
+{
+	struct mt6397_rtc *rtc = dev_get_drvdata(priv);
+	unsigned int ival;
+	int ret;
+	u8 *buf = val;
+
+	mutex_lock(&rtc->lock);
+
+	for (; bytes; bytes--) {
+		ival = *buf++;
+		ret = regmap_field_write(rtc->spare[offset++], ival);
+		if (ret)
+			goto out;
+	}
+	mtk_rtc_write_trigger(rtc);
+out:
+	mutex_unlock(&rtc->lock);
+	return ret;
+}
+
 static irqreturn_t mtk_rtc_irq_handler_thread(int irq, void *data)
 {
 	struct mt6397_rtc *rtc = data;
@@ -244,6 +296,42 @@
 	.set_alarm  = mtk_rtc_set_alarm,
 };
 
+static int mtk_rtc_set_spare(struct device *dev)
+{
+	struct mt6397_rtc *rtc = dev_get_drvdata(dev);
+	struct reg_field tmp[SPARE_RG_MAX];
+	int i, ret;
+	struct nvmem_config nvmem_cfg = {
+		.name = "mtk_rtc_nvmem",
+		.word_size = SPARE_REG_WIDTH,
+		.stride = 1,
+		.size = SPARE_RG_MAX * SPARE_REG_WIDTH,
+		.reg_read = rtc_nvram_read,
+		.reg_write = rtc_nvram_write,
+		.priv = dev,
+	};
+
+	memcpy(tmp, rtc->data->spare_reg_fields, sizeof(tmp));
+
+	for (i = 0; i < SPARE_RG_MAX; i++) {
+		tmp[i].reg += rtc->addr_base;
+		rtc->spare[i] = devm_regmap_field_alloc(rtc->rtc_dev->dev.parent,
+							rtc->regmap,
+							tmp[i]);
+		if (IS_ERR(rtc->spare[i])) {
+			dev_err(rtc->rtc_dev->dev.parent, "spare regmap field[%d] err= %ld\n",
+						i, PTR_ERR(rtc->spare[i]));
+			return PTR_ERR(rtc->spare[i]);
+		}
+	}
+
+	ret = devm_rtc_nvmem_register(rtc->rtc_dev, &nvmem_cfg);
+	if (ret)
+		dev_err(rtc->rtc_dev->dev.parent, "nvmem register failed\n");
+
+	return ret;
+}
+
 static int mtk_rtc_probe(struct platform_device *pdev)
 {
 	struct resource *res;
@@ -294,6 +382,10 @@
 	rtc->rtc_dev->start_secs = mktime64(1968, 1, 2, 0, 0, 0);
 	rtc->rtc_dev->set_start_time = true;
 
+	if (rtc->data->spare_reg_fields)
+		if (mtk_rtc_set_spare(&pdev->dev))
+			dev_err(&pdev->dev, "spare is not supported\n");
+
 	return devm_rtc_register_device(rtc->rtc_dev);
 }
 
@@ -324,6 +416,7 @@
 
 static const struct mtk_rtc_data mt6358_rtc_data = {
 	.wrtgr = RTC_WRTGR_MT6358,
+	.spare_reg_fields = mtk_rtc_spare_reg_fields,
 };
 
 static const struct mtk_rtc_data mt6397_rtc_data = {
