--- a/drivers/iio/adc/mt6359-auxadc.c	2026-09-27 16:55:51.062099118 -0500
+++ b/drivers/iio/adc/mt6359-auxadc.c	2026-09-27 16:55:30.703001032 -0500
@@ -155,7 +155,7 @@
 		.endianness = IIO_CPU						\
 	},									\
 	.indexed = 1,								\
-	.info_mask_separate = BIT(IIO_CHAN_INFO_RAW) | BIT(IIO_CHAN_INFO_SCALE)	\
+	.info_mask_separate = BIT(IIO_CHAN_INFO_PROCESSED),			\
 }
 
 static const struct iio_chan_spec mt6357_auxadc_channels[] = {
@@ -473,7 +473,7 @@
 	/* Stop sampling */
 	regmap_write(regmap, cinfo->regs[desc->req_idx], 0);
 
-	*out = val & GENMASK(chan->scan_type.realbits - 1, 0);
+	*out = val;
 	return 0;
 }
 
@@ -492,17 +492,6 @@
 	const struct mtk_pmic_auxadc_chan *desc = &cinfo->desc[chan->scan_index];
 	int ret;
 
-	if (mask == IIO_CHAN_INFO_SCALE) {
-		*val = desc->r_ratio.numerator * AUXADC_VOLT_FULL;
-
-		if (desc->r_ratio.denominator > 1) {
-			*val2 = desc->r_ratio.denominator;
-			return IIO_VAL_FRACTIONAL;
-		}
-
-		return IIO_VAL_INT;
-	}
-
 	scoped_guard(mutex, &adc_dev->lock) {
 		switch (chan->scan_index) {
 		case PMIC_AUXADC_CHAN_IBAT:
@@ -517,6 +506,8 @@
 		}
 	}
 
+	*val &= GENMASK(chan->scan_type.realbits - 1, 0);
+
 	if (ret) {
 		/*
 		 * If we get more than one timeout, it's possible that the
@@ -533,6 +524,14 @@
 	}
 	adc_dev->timed_out = false;
 
+	if (mask == IIO_CHAN_INFO_PROCESSED) {
+		*val *= desc->r_ratio.numerator * AUXADC_VOLT_FULL;
+		if (desc->r_ratio.denominator > 1) {
+			*val /= desc->r_ratio.denominator;
+		}
+		*val >>= chan->scan_type.realbits;
+	}
+
 	return IIO_VAL_INT;
 }
 
